### PR TITLE
Issue 1776: Refactor PipelineRunManagerTest. Second Iteration.

### PR DESCRIPTION
**Pull request by Sergeev-Ilya**
Issue: https://github.com/epam/cloud-pipeline/issues/1776

### CODE REVIEW

File: api/src/test/java/com/epam/pipeline/test/creator/pipeline/PipelineCreatorUtils.java [hunk 1]
```diff
@@ -119,6 +124,21 @@ public static PipelineRun getPipelineRun(final Long id, final String owner) {
         return pipelineRun;
     }
 
+    public static PipelineRun getPipelineRunWithInstance(final Long id, final String owner, final Long runInstanceId) {
+        final PipelineRun pipelineRun = getPipelineRun(id, owner);
+        final RunInstance parentRunInstance = new RunInstance();
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/test/creator/pipeline/PipelineCreatorUtils.java:136]: Please remove `parent` prefix.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 2]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+        assertEquals(2, projectFilter.getPipelineIds().size());
+        assertEquals(2, projectFilter.getConfigurationIds().size());
+    }
+
+    @Test
+    public void shouldThrowExceptionOnInexistentRunTagUpdate() {
+        doReturn(null).when(pipelineRunDao).loadPipelineRun(eq(NOT_EXISTING_RUN_ID));
+
+        final Runnable result = () -> pipelineRunManager.updateTags(NOT_EXISTING_RUN_ID, new TagsVO(null));
+        assertThrows(IllegalArgumentException.class, result);
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:243]: I think it's okay to inline the lambda variable.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 3]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+    @Test
+    public void shouldReplaceEnvVarsWithTestEnvWithSeveralVariables() {
+        final Map<String, String> envVars = new HashMap<>();
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
+
+        final String paramValue = String.format("test/$%s/${%s}/$%s/", ENV_VAR_NAME, ENV_VAR_NAME, ENV_VAR_NAME);
+        final String expectedValue = String.format("test/%s/%s/%s/", ENV_VAR_VALUE, ENV_VAR_VALUE, ENV_VAR_VALUE);
+        final List<PipelineRunParameter> actualParameters = pipelineRunManager.replaceParametersWithEnvVars(
+                Collections.singletonList(new PipelineRunParameter(PARAM_NAME_1, paramValue)), envVars);
+        checkResolvedValue(actualParameters, paramValue, expectedValue);
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:309]: You can replace all of this with `testEnvVarsReplacement` call if you use the following formatting syntax `%1$s`  in the patterns.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 4]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+    }
+
+    private void testEnvVarsReplacement(final String paramValuePattern, final String expectedValuePattern) {
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:312]: Let's call these method `assertEnvVarsReplacement` similarly to other private methods in the class.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 5]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
+
+        final String paramValue = String.format(paramValuePattern, ENV_VAR_NAME);
+        final String expectedValue = String.format(expectedValuePattern, ENV_VAR_VALUE);
+        final List<PipelineRunParameter> actualParameters = pipelineRunManager.replaceParametersWithEnvVars(
+                Collections.singletonList(new PipelineRunParameter(PARAM_NAME_1, paramValue)), envVars);
+        checkResolvedValue(actualParameters, paramValue, expectedValue);
+    }
+
+    private void checkResolvedValue(List<PipelineRunParameter> actualParameters, String paramValue,
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:323]: Let's inline this method.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 6]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+        final Runnable result = () -> pipelineRunManager.updateTags(NOT_EXISTING_RUN_ID, new TagsVO(null));
+        assertThrows(IllegalArgumentException.class, result);
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyCollections() {
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(new ArrayList<>(), new HashMap<>());
+        assertTrue(CollectionUtils.isEmpty(actualParameters));
+    }
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:251]: All `shouldReplaceEnvVars` methods have the same structure. We have:
- initial parameters list
- env vars
- expected parameters list

It looks like we can use a single method to implement all these tests
```java
assertEnvVarsReplacement(Collections.emptyList(), # initial parameters
    Collections.emptyList(), # expected parameters
    Collections.emptyMap()) # env vars
```

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 7]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyEnvVars() {
+        final List<PipelineRunParameter> parameters = new ArrayList<>();
+        parameters.add(new PipelineRunParameter(PARAM_NAME_1, TEST_STRING));
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:256]: Use `Collections.singletonList` for such list initialization.

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java [hunk 8]
```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
...
+
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(parameters, new HashMap<>());
+        assertEquals(parameters, actualParameters);
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyParams() {
+        final Map<String, String> envVars = new HashMap<>();
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
```
Comment (tcibinan) [inline api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:266]: And here use `Collections.singletonMap` for such map initialization.

**Review by tcibinan — CHANGES_REQUESTED** [review]

File: api/src/test/java/com/epam/pipeline/test/creator/pipeline/PipelineCreatorUtils.java (hunk 1 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/test/creator/pipeline/PipelineCreatorUtils.java:136]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 2 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:243]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 3 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:309]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 4 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:312]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 5 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:323]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 6 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:251]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 7 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:256]: Done!

File: api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java (hunk 8 above)
Comment (Sergeev-Ilya) [reply api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java:266]: Done!

**Review by Sergeev-Ilya — COMMENTED** [review]
I have decided to create a new PR to develop, adding all the code from this (current PR), including all refactoring advices. The new PR is: https://github.com/epam/cloud-pipeline/pull/1841

After new PR will de merged, this will be deleted.

**Comment by Sergeev-Ilya** [comment]
I have decided to create a new PR to develop, adding all the code from this (current PR), including all refactoring advices. The new PR is: #1841

After new PR will de merged, this will be deleted.

**Pull request closed without merging**